EIPS/eip-5792.md

```diff
@@ -31,42 +31,77 @@ In updating to a new set of `wallet_` RPCs, our main goals are to enforce a clea
 Four new JSON-RPC methods are added: three are for handling batches of onchain calls, and one is for querying support for wallet capabilities, such as to make better use of the three batching methods.
 Apps may begin using these first three methods immediately, falling back to `eth_sendTransaction` and `eth_getTransactionReceipt` when they are not available.
 
-We also define one capability expression for use with the fourth method, which further enables a wallet to promise atomicity of the execution of calls passed and managed by the first three methods.
-
 ### `wallet_sendCalls`
 
-Requests that a wallet submits a batch of calls. `from` and `chainId` are top-level properties rather than per-call properties because all calls should be sent from the same sender and on the same chain, identified by [EIP-155](./eip-155.md) integers expressed in hexadecimal notation. The items in the `calls` field are only those that are shared by all transaction types. Any other fields that a wallet may need to submit a transaction should be handled by the wallet.
+Requests that a wallet submits a batch of calls. `from` and `chainId` are identified by [EIP-155](./eip-155.md) integers expressed in hexadecimal notation,
+with `0x` prefix and no leading zeroes for the `chainId` value.
+The items in the `calls` field are simple `{to, data, value}` tuples.
 
 The capabilities field is how an app can communicate with a wallet about capabilities a wallet supports. For example, this is where an app can specify a paymaster service URL from which an [ERC-4337](./eip-4337.md) wallet can request a `paymasterAndData` input for a user operation.
 
+Each capability defined in the "capabilities" member can define global or call specific fields.
+These fields are set inside this capability's entry in the `capabilities` object.
+Each entity in the `calls` field may contain an optional `capabilities` object.
+This object allows the applications to attach a capability-specific metadata to individual calls.
+
+Unless these requirements are explicitly overriden by a certain `capability`, the wallet must adhere to the following rules.
+Note that such a `capability` is not in the scope of this EIP and should be defined in its own ERC.
+
 The wallet:
 
 * MUST send the calls in the order specified in the request
-* MUST send the calls on the same chain identified by the call's `chainId` property
+* MUST send the calls on the same chain identified by the request's `chainId` property
+* MUST send the calls from the address specified in the request's `from` property, if provided
+  * If `from` is not provided the wallet SHOULD present the user with an opportunity to view and change the `from` address during confirmation
```

Comment:
This is a little hard to parse-- if it's not provided there's nothing to change? do you mean:
```suggestion
  * If `from` is not provided the wallet SHOULD present the user with an opportunity to select a `from` address during confirmation
```

Reply:
Replaced 'change' with 'select' to indicate this field is not user-editable